car/car.ino

```diff
@@ -129,3 +122,71 @@ void handleInput() { //handle serial input (String!!)
     }  
   }
 }
+
+void driveAroundObstical(){
+  int leftDistance = left.readRangeContinuousMillimeters();
+  int rightDistance = right.getMedianDistance();
+  
+  if (leftDistance > LEFT_DIST && rightDistance > RIGHT_DIST){               //if there is no obstical on either sides rotate 90 degrees to the right
+    obsticalTurnRight();
+  }  
+  else if (rightDistance <= RIGHT_DIST && rightDistance > 0){               //Turn left if obstical on the right   
+    obsticalTurnLeft();
+  }
+  else if (leftDistance <= LEFT_DIST && leftDistance > 0){                  //Turn right if obstical on the left side
+    obsticalTurnRight(); 
+  }
+  else{
+    rotateOnSpot(TURN_AROUND, TURN_SPEED);                                  //it rotates 180 degrees to the right.
+  } 
+}
+
+void obsticalTurnRight(){  //this method makes the car go around an obstical on the right side
+    rotateOnSpot(TURN_RIGHT, TURN_SPEED);
+    leftOdometer.reset();
+    rightOdometer.reset();
+
+    while(left.readRangeContinuousMillimeters() <= LEFT_DIST && left.readRangeContinuousMillimeters() > 0){      //while there is still an obstical on the left side continue forward
+         forward(TURN_SPEED);
+    }
+    goDistance(15, TURN_SPEED);
+    stop();
+    const long odometerLength = ((leftOdometer.getDistance() + rightOdometer.getDistance())/2);                 //this distance it the distance the car moved sideways
+    
+    rotateOnSpot(TURN_LEFT, TURN_SPEED);
+    goDistance(60, TURN_SPEED);  
+       
+    while(left.readRangeContinuousMillimeters() <= LEFT_DIST && left.readRangeContinuousMillimeters() > 0){       //while there is still an obstical on the left side continue forward
+         forward(TURN_SPEED);
+    }
+    stop();
+    
+    rotateOnSpot(TURN_LEFT, TURN_SPEED);
+    goDistance(odometerLength, TURN_SPEED);             //this distance it the distance the car moved sideways (used to go back in the same path but on the other side of the obstical.)
+    rotateOnSpot(TURN_RIGHT, TURN_SPEED);
+}
+
+void obsticalTurnLeft(){    //this method makes the car go around an obstical on the left side
+    rotateOnSpot(TURN_LEFT, TURN_SPEED); 
+    leftOdometer.reset();
+    rightOdometer.reset();
+
+    while(right.getMedianDistance() <= RIGHT_DIST && right.getMedianDistance() > 0){                              //while there is still an obstical on the right side continue forward
```

Comment:
Replace the calls to `getMedianDistance` as instructed in the pull request. Same below.